torch/csrc/jit/script/compilation_unit.h

```diff
@@ -283,5 +283,17 @@ struct StrongFunctionPtr {
   std::shared_ptr<script::CompilationUnit> cu_;
   Function* function_;
 };
+
+class ScriptCodeObj : public StrongFunctionPtr {
```

Comment:
This inheritance doesn't make sense because ScriptCodeObj is not a function pointer. We should make ScriptCodeObj a separate type from StrongFunctionPtr and then bind its `__call__` function to create an instance of the object rather than construct a script method whose purpose is to do this. Otherwise it is too easy to accidentally use ScriptCodeObj as a function.

Also naming: ScriptCodeObj is not an Obj, it is a class. It is also not Code. ScriptClass might be more accurate.